test/functional/change_stream.test.js

```diff
@@ -1824,77 +1828,46 @@ describe('Change Streams', function () {
       }
     });
 
-    it(
-      'should throw MongoDriverError when set as an emitter with "on" and used as an iterator with "hasNext"',
-      {
-        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
-        test: async function () {
-          await new Promise(resolve => changeStream.on('change', resolve));
-          try {
-            await changeStream.hasNext().catch(err => {
-              expect.fail(err.message);
-            });
-          } catch (error) {
-            return expect(error).to.be.instanceof(MongoDriverError);
-          }
-          return expect.fail('Should not reach here');
-        }
-      }
-    );
-
-    it(
-      'should throw MongoDriverError when set as an iterator with "hasNext" and used as an emitter with "on"',
-      {
-        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
-        test: async function () {
-          await changeStream
-            .hasNext()
-            .catch(() => expect.fail('Failed to set changeStream to iterator'));
-          try {
-            await new Promise(resolve => changeStream.on('change', resolve));
-          } catch (error) {
-            return expect(error).to.be.instanceof(MongoDriverError);
-          }
-          return expect.fail('Should not reach here');
-        }
-      }
-    );
-
-    it(
-      'should throw MongoDriverError when set as an emitter with "once" and used as an iterator with "next"',
-      {
-        metadata: { requires: { topology: 'replicaset', mongodb: '>=3.6' } },
-        test: async function () {
-          await new Promise(resolve => changeStream.once('change', resolve));
-          try {
-            await changeStream.next().catch(err => {
-              expect.fail(err.message);
-            });
-          } catch (error) {
-            return expect(error).to.be.instanceof(MongoDriverError);
-          }
-          return expect.fail('Should not reach here');
-        }
+    it(`should throw when mixing event listeners with iterator methods`, {
```

Comment:
Not sure why the deps update trigger this, but the interval style of testing here was causing flakiness that I couldn't get passed, did a little refactor to clear the issues up.